libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -104,7 +104,7 @@ AP_BattMonitor::init()
 #ifdef HAL_BATT_MONITOR_DEFAULT
     if (_params[0]._type == 0) {
         // we can't use set_default() as the type is used as a flag for parameter conversion
-        _params[0]._type.set((AP_BattMonitor_Params::BattMonitor_Type)HAL_BATT_MONITOR_DEFAULT);
+        _params[0]._type.set(float(HAL_BATT_MONITOR_DEFAULT));
```

Comment:
`_type` is `AP_Int8` why cast to float?

Reply:
Can't pass the enumeration type into the `set` method.

We pass both the enumeration type *and* flat values in as that macro value.

Reply:
.... but there's still the question as to "float" rather than Int.  Reflex, probably.

Reply:
Fixed.